apps/src/dance/ai/DanceAiModal.tsx

```diff
@@ -165,6 +151,12 @@ const DanceAiModal: React.FunctionComponent<DanceAiModalProps> = ({
     (state: {dance: DanceState}) => state.dance.aiOutput
   );
 
+  const currentInputSlot = inputs.length;
+  const selectedEmojis = useMemo(
+    () => inputs.map(getItem).filter(item => item !== undefined),
+    [inputs]
+  );
+
```

Comment:
by memoizing `selectedEmojis` we can save a few render steps + not need to look up the item every time.